frontend/src/types.ts

```diff
@@ -58,10 +65,12 @@ export type Scope =
   | "sign_message"
   | "notifications"; // covers all notification types
 
+export type ScopeGroupType = "send_receive" | "only_receive" | "custom";
+
 export type Nip47NotificationType = "payment_received" | "payment_sent";
 
 export type IconMap = {
-  [key in Scope]: LucideIcon;
+  [key in string]: LucideIcon;
```

Comment:
Now there is no type safety and we will not know when we introduce a new scope. Can you undo the change and add a `ScopeIconMap` and any `OtherIconMap` that is typed

Reply:
I just skipped it here as it was causing issues (or could've added icon maps separately for methods and scopes) Should I proceed with the later?